src/ConstructiveSolidGeometry/plotting/CSG/CSG.jl

```diff
@@ -1,37 +1,24 @@
-@recipe function f(csg::ConstructiveSolidGeometry.CSGUnion)
-    linecolor --> :black
-    csguniontype --> :union
-    @series begin 
-        linestyle --> :solid
-        csg.a
-    end
-    @series begin 
-        linestyle --> :solid
-        csg.b
-    end
+primitives(vp::AbstractVolumePrimitive) = (vp,)
+function primitives(csg::AbstractConstructiveGeometry)
+    ps = []
+    push!(ps, primitives(csg.a)...)
+    push!(ps, primitives(csg.b)...)
+    ps
 end
-@recipe function f(csg::ConstructiveSolidGeometry.CSGDifference)
-    linecolor --> :black
+
+@recipe function f(csg::AbstractConstructiveGeometry)
+    ps = primitives(csg)
     @series begin
-        linestyle --> :solid
-        csguniontype --> :difference
-        csg.a
+        label --> "CSG"
+        linestyle := isClosedPrimitive(ps[1]) ? :solid : :dot
```

Comment:
So this means, that `OpenPrimitives` are always the negative geometry of a `CSGDifference`?

Reply:
No. Differences of Differences switches Open an Closed and so on...
There is no ideal solution for plotting this. This is just a simple way to have a look at the CSG.
But for sure nothing for a publication.